geomstats/datasets/sample_sdp_2d.py

```diff
@@ -0,0 +1,131 @@
+
+import geomstats.backend as gs
+from geomstats.geometry.spd_matrices import SPDMatrices
+from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+
+
+class DatasetSPD_2D():
+    """
+    Sample 2D SPD dataset.
+
+    Data is of shape [n_samples * n_classes, n_features];
+    Labels are of shape [n_samples * n_classes, n_classes].
+    """
+
+    def __init__(self, n_samples=100, n_features=2, n_classes=3):
+        self.n_samples = n_samples
+        self.n_features = n_features
+        self.n_classes = n_classes
+        self.data_helper = DataHelper()
+
+    def generate_sample_dataset(self):
+        """
+        Generate the dataset.
+
+        :return:
+        X: array-like, shape = [n_samples * n_classes, n_features]: data
+        Y: array-like, shape = [n_samples * n_classes, n_classes]: labels
+        """
+        X, Y = self.setup_data()
+        X, Y = self.data_helper.shuffle(X, Y)
+        return X, Y
+
+    def setup_data(self):
+        """
+        Generate the un-shuffled dataset.
+
+        :return:
+        X: array-like, shape = [n_samples * n_classes, n_features]: data
+        Y: array-like, shape = [n_samples * n_classes, n_classes]: labels
+        """
+        mean_covariance_eigenvalues = gs.random.uniform(
+            0.1, 5., (self.n_classes, self.n_features))
+        base_rotations = SpecialOrthogonal(n=self.n_features).random_gaussian(
+            gs.eye(self.n_features), 1, n_samples=self.n_classes)
+        # var_eigenvalues = gs.random.uniform(
+        #     .04, .06, (self.n_classes, self.n_features))
+        var_rotations = gs.random.uniform(
+            .5, .75, (self.n_classes))
+
+        # data
+        cov = gs.zeros(
+            (self.n_classes *
+             self.n_samples,
+             self.n_features,
+             self.n_features))
+        Y = gs.zeros((self.n_classes * self.n_samples, self.n_classes))
+        for i in range(self.n_classes):
+            cov[i * self.n_samples:(i + 1) * self.n_samples] = self.make_data(
+                base_rotations[i], gs.diag(
+                    mean_covariance_eigenvalues[i]), var_rotations[i])
+            Y[i * self.n_samples:(i + 1) * self.n_samples, i] = 1
+        return cov, Y
+
+    def make_data(self, eigenspace, eigenvalues, var):
+        """
+        Generate Gaussian data from mean matrix and variance.
+
+        :param eigenspace: array-like, shape = [n, n]
+        :param eigenvalues: array-like, shape = [n, n] (diagonal matrix)
+        :param var: float, variance of the wanted distribution.
+        :return: array-like, shape = [n, n]: data.
+        """
+        spd = SPDMatrices(n=self.n_features)
+        spd.set_eigensummary(eigenspace, eigenvalues)
+        spd_data = spd.random_gaussian(
+            var_rotations=var, n_samples=self.n_samples)
+        return spd_data
+
+    def make_data_noisy(self, eigenspace, eigenvalues, var, var_eigenvalues):
```

Comment:
This method is not used?

Reply:
I think there is no need for `make_data` and `make_data_noisy`, since they are just calling the spd mathods: the `spd.random_gaussian` could directly be called in `setup_data`?

Reply:
It is an option to use it, instead of the not-noisy one. In the illustration example, I use `make_data`, and I had left the usage of `make_data_noisy` in comments (since then deleted).
I found the `make_data` methods were convenient to wrap a few lines of code together. The setup defines hyper-parameters, the making creates the data, calling the class sampler.

Reply:
Ok!